### PR TITLE
Sk rst

### PR DESCRIPTION
**Pull request by yuval-k**
this is still not done. but rst generation is under a flag so generation shouldn't change in the mean time.
opening a PR as this might take me a bit more time, and i want to make sure the foundations are in

### CODE REVIEW

File: pkg/code-generator/docgen/options/options.go [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package options
+
+type DocsOutput string
+
+const (
+	Markdown     DocsOutput = "markdown"
```
Comment (rickducott) [inline pkg/code-generator/docgen/options/options.go]: I'm confused by this const block

Comment (yuval-k) [reply pkg/code-generator/docgen/options/options.go]: this is an 'Enum'

**Comment by rickducott** [comment]
Unclear why the CI build started failing, specifically with this error:

```
• Failure [12.896 seconds]
Step #2 - "test": Test Kube ResourceClient
Step #2 - "test": /workspace/gopath/src/github.com/solo-io/solo-kit/pkg/api/v1/clients/kube/resource_client_test.go:38
Step #2 - "test": multiple namespaces
Step #2 - "test": /workspace/gopath/src/github.com/solo-io/solo-kit/pkg/api/v1/clients/kube/resource_client_test.go:109
Step #2 - "test": can watch resources across namespaces when using NamespaceAll [It]
Step #2 - "test": /workspace/gopath/src/github.com/solo-io/solo-kit/pkg/api/v1/clients/kube/resource_client_test.go:144
Step #2 - "test": 
Step #2 - "test": drained the watch channel 50 times, something is wrong
```

**Comment by rickducott** [comment]
**Passes** against local minikube (Kube 1.13.0):
`minikube start --memory=8192 --cpus=4 --kubernetes-version=v1.13.0`

**Passes** against local minikube (Kube 1.11.5):
`minikube start --memory=8192 --cpus=4 --kubernetes-version=v1.11.5`

**Fails** against GKE test cluster (Kube 1.11.5):
`gcloud container clusters get-credentials --project=solo-public --zone=us-central1-a test-cluster`

**Comment by rickducott** [comment]
Discovered some resources hanging around the cluster, deleted them manually and the tests passed locally against GKE

**Comment by rickducott** [comment]
Filed https://github.com/solo-io/solo-kit/issues/85 to follow up on the strange test failure / fix

**Pull request merged**